lib/std/io/reader.zig

```diff
@@ -281,28 +281,28 @@ pub fn Reader(
 
         /// Reads a native-endian integer
         pub fn readIntNative(self: Self, comptime T: type) !T {
-            const bytes = try self.readBytesNoEof((@typeInfo(T).Int.bits + 7) / 8);
+            const bytes = try self.readBytesNoEof(@intCast(u16, (@as(u17, @typeInfo(T).Int.bits) + 7) / 8));
```

Comment:
We could of course actually add a helper like this to std.builtin.Int:
```zig
    /// This data structure is used by the Zig language code generation and
    /// therefore must be kept in sync with the compiler implementation.
    pub const Int = struct {
        signedness: Signedness,
        bits: u16,

        fn byte_count(int: Int) @Int(0, 8192) {
            return @intCast((@as(u17, int.bits) + 7) / 8);
        }
    };
```
This would certainly make this clearly common and easy-to-make mistake harder to make.
(I used *futuristic* Zig here because I felt like it but ignore that)

Reply:
Naive question: does `@sizeOf` not provide the correct result for every `T`?

Reply:
Note that `std.math.divCeil(int.bits, 8)` seems to handle this case correctly.  Personally, I would be in favor of a `@divCeil` builtin that also avoids this overflow condition.

Reply:
Oh that sounds interesting actually. I finished writing a short proposal just now just to get some feedback on your idea but then I found #4095. Should I still open it you think? I don't actually see much discussion on why specifically there shouldn't be a builtin for `@divCeil` other than maybe this:

> It should only be a builtin if this is something that has a matching instruction on some CPU. Otherwise it just goes into the standard library.

My argument here might be that you will never know what kind of instructions CPUs might have in the future, though. And maybe now a CPU has an instruction for it?

Reply:
Either way I think I will leave this PR as-is for now though. That was just a little side commit to fix this problem at least a bit and there are certainly more of these issues lingering around so the improvement of using `std.math.divCeil` can be done in a future PR that focuses more on that.

Reply:
I'm also not aware of a cpu with a `@divFloor` instruction, `@divCeil` would end up being lowered in a very similar way.

Reply:
I'm also not aware of a cpu with a `@divFloor` instruction, `@divCeil` would end up being lowered in a very similar way. As a backend writer, I'm always in favor of more builtins, not less (currently really missing `@abs`).

Reply:
Yeah, I think `@abs` would be kinda nice too.